### PR TITLE
docs: docs, docs, docs!

### LINKED ISSUE #31: docs, docs, docs!
tons of examples. this tool is only useful if people can use it with minimal overhead, otherwise people will revert to doing their own setups.

### PR DESCRIPTION
**Pull request by pinheadmz**
Closes https://github.com/bitcoin-dev-project/warnet/issues/31

Experience it here: https://github.com/bitcoin-dev-project/warnet/tree/docs-docs-docs

### CODE REVIEW

**Comment by willcl-ark** [comment]
Love it!

edit: yuck, americanised spellings 🤢

File: docs/warcli.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+the network you want to control. This is set by default to `--network="warnet"`
+to simplify default operation.
+
+Execute `warcli --help` or `warcli help` to see a list of command categories.
+
+`warcli` commands are organized in a hierarchy of categories and subcommands:
+
+```
+warcli
+  + help       Display help information for a given command.
```
Comment (willcl-ark) [inline docs/warcli.md:17]: Is this auto-generated somehow?

**Comment by pinheadmz** [comment]
> edit: yuck, americanised spellings 🤢

war ain't pretty.

also
TODO:
- [ ] data_collection.md (grafana, log grepper, p2p messages)
- [ ] gui.md (@Extheoisah )
- [ ] architecture.md (explain host, docker, warnet server, tanks)

**Review by willcl-ark — APPROVED** [review]
OG changes ACK from me.

I added 3 more commits with a few small additions for your review, happy to drop them if you just want to merge this and don't like them.

Do we want to address the additional sections in a followup; I think that makes sense as this looks good-to-go IMO.

@josibake any comments?

File: docs/warcli.md (hunk 1 above)
Comment (pinheadmz) [reply docs/warcli.md:17]: no i typed it up manually and im not sure i love it. could be a bit annoying to maintain

**Pull request merged**